### PR TITLE
Replace highlight.js with shiki

### LINKED ISSUE #1290: tsx and highlightjs
I couldn't figure out how to get `tsx` code to highlight properly. What am I missing? Highlighjs seems to support tsx out of the box. 🤷🏻‍♂️

### PR DESCRIPTION
**Pull request by maximkott**
fixes #1290
Working typescript demo here: https://maximkott.github.io/typedoc-default-themes/, needs way more testing though.

### CODE REVIEW

**Comment by WilliamJns** [comment]
+1 
This looks to be the most reliable and easiest way to improve code for now.

File: tsconfig.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
       "es2015.core",
       "es2015.collection",
       "es2015.iterable",
-      "es2016.array.include" // Supported by Node 6+
+      "es2016.array.include", // Supported by Node 6+
```
Comment (krisztianb) [inline tsconfig.json:9]: Doesn't this conflict with the target of "es2015"? As far as I can tell, we have no Babel in this project.

**Comment by krisztianb** [comment]
Hi there. Nice job. In the PR title you mention prism.js but you seem to include shiki instead.

File: tsconfig.json (hunk 1 above)
Comment (maximkott) [reply tsconfig.json]: I've removed the *dom* lib from tsconfig, this was a mistake. It was required for prismjs, but not necessary for shiki.

**Comment by maximkott** [comment]
> Hi there. Nice job. In the PR title you mention prism.js but you seem to include shiki instead.

Hi and thanks! It was meant to be used with prismjs, but then someone proposed to use shiki instead: https://github.com/TypeStrong/typedoc/issues/1290#issuecomment-623048810, so I switched one for another. I'll fix the title.

**Comment by Gerrit0** [comment]
0.20.0-beta.19 has switched to Shiki - sorry this took so long to do! Due to the conflicts, it was easier to essentially restart this work. I also found some issues with the chosen languages that required extra logic.

**Comment by krisztianb** [comment]
So this PR could be receted then?

**Comment by Gerrit0** [comment]
I'm planning to leave it open until release to remind me to thank maximkott in the release notes

**Comment by Gerrit0** [comment]
0.20 has been released, which includes these changes. Thanks again @maximkott!

**Pull request closed without merging**